quill/src/detail/misc/Os.cpp

```diff
@@ -156,6 +156,7 @@ void set_thread_name(char const* name)
 {
 #if defined(__CYGWIN__)
   // set thread name on cygwin not supported
+  (void)name;
 #elif defined(__MINGW32__) || defined(__MINGW64__)
   // Disabled on MINGW.
   (void)name;
```

Comment:
Hey @odygrd! I think you could condense these (feel free to edit):
```suggestion
#if defined(__CYGWIN__) || defined(__MINGW32__) || defined(__MINGW64__)
  // Disabled/not supported on Cygwin and MinGW.
  (void)name;
```